internal/injector/aspect/context/golang.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package context
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v3"
+)
+
+// GoLang represents a go language level. It's a string of the form "go1.18".
+type GoLang struct {
+	semver string
+}
+
+func ParseGoLang(lang string) (GoLang, error) {
+	// Validate the language level value...
+	if !strings.HasPrefix(lang, "go1.") {
+		return GoLang{}, fmt.Errorf(`invalid go language level (expected e.g, "go1.18"): %q`, lang)
+	}
+	for _, r := range lang[4:] {
+		if !unicode.IsDigit(r) {
+			return GoLang{}, fmt.Errorf(`invalid go language level (expected e.g, "go1.18"): %q`, lang)
+		}
```

Comment:
the package `go/version` has some utilities to work with go version strings.

So probably you can replace this code with just `version.IsValid(lang)`

Reply:
Ohhhhhhhh :)